### ISSUE TITLE
Code 30, reason: unknown error

### ISSUE THREAD

**Issue by dyslexicjedi** [ageitgey/face_recognition#358]
* face_recognition version: 1.2.1
* Python version: 3.5.2
* Operating System: Linux Mint

### Description

Attempting to run the face_recognition in anyway, changed directories and without directories, all produce same issue.

### What I Did

rob@Turing ~/face_recognition $ face_recognition /known/ /unknown/

Traceback (most recent call last):
  File "/usr/local/bin/face_recognition", line 7, in <module>
    from face_recognition.cli import main
  File "/usr/local/lib/python3.5/dist-packages/face_recognition/__init__.py", line 7, in <module>
    from .api import load_image_file, face_locations, batch_face_locations, face_landmarks, face_encodings, compare_faces, face_distance
  File "/usr/local/lib/python3.5/dist-packages/face_recognition/api.py", line 24, in <module>
    cnn_face_detector = dlib.cnn_face_detection_model_v1(cnn_face_detection_model)
RuntimeError: Error while calling cudaMallocHost(&data, new_size*sizeof(float)) in file /tmp/pip-build-_n7xryvl/dlib/dlib/dnn/gpu_data.cpp:185. code: 30, reason: unknown error

**Comment by ageitgey** [comment]
The most likely cause is probably a bad installation of CUDA. Make sure the CUDA kernel module running is the same version as the CUDA library used to compile dlib.

**Comment by dyslexicjedi** [comment]
I'm running cuda 9.0.176 (downloaded from nvidia), the dlib was installed via the pip3 command (19.7) from your installation guide.  I'll try to compile dlib locally and see if that fixes the issue.

**Comment by dyslexicjedi** [comment]
Nope, same issue. I uninstalled dlib via pip3. Then build dlib from source with cuda support enabled. and I'm getting the same code 30 error when running face_recognition.

**Comment by dyslexicjedi** [comment]
On a whim, I decided to run face_recognition via the sudo command and it worked. It must have been a permissions issue of some kind, not sure. but hey, if it works, it works.

**Comment by julayelei** [comment]
Fixed using sudo

**Comment by guesswh0** [comment]
got same problem while running raw `dlib.cnn_face_detection_model_v1('mmod_human_face_detector.dat')`

`RuntimeError: Error while calling cudaGetDevice(&the_device_id) in file /tmp/pip-install-6k2tx_yh/dlib/dlib/cuda/gpu_data.cpp:178. code: 30, reason: unknown error`

fixed just by restarting computer ;)

**Comment by ivineetm007** [comment]
Hi,
My system got stuck in login loop and to fix it I removed nvidia-drivers and then reinstalled it.
After that I am facing error:
```
Traceback (most recent call last):
  File "face_clustering.py", line 282, in <module>
    facerec = dlib.face_recognition_model_v1(face_rec_model_path)
RuntimeError: Error while calling cudaMallocHost(&data, new_size*sizeof(float)) in file /tmp/pip-install-usd7a31_/dlib/dlib/cuda/gpu_data.cpp:185. code: 30, reason: unknown error
```
I restarted the pc, reinstalled dlib, cudatoolkit in the conda environment but nothing works.

System details:
Ubuntu 16.04
Python version - 3.5.5 in the conda environment.
conda environment:
<details>
absl-py                   0.4.1                    py35_0    defaults
alabaster                 0.7.10           py35h6fb19ab_0    defaults
anaconda                  custom           py35h104c396_0    defaults
anaconda-client           1.6.9                    py35_0    defaults
anaconda-project          0.8.2            py35h7d62c80_0    defaults
asn1crypto                0.24.0                   py35_0    defaults
astor                     0.7.1                    py35_0    defaults
astroid                   1.6.1                    py35_0    defaults
astropy                   2.0.3            py35h14c3975_0    defaults
attrs                     17.4.0                   py35_0    defaults
babel                     2.5.3                    py35_0    defaults
backports                 1.0              py35hd471ac7_1    defaults
backports.shutil_get_terminal_size 1.0.0            py35h40844db_2    defaults
beautifulsoup4            4.6.0            py35h442a8c9_1    defaults
bitarray                  0.8.1            py35h14c3975_1    defaults
bkcharts                  0.2              py35he4f7e30_0    defaults
blas                      1.0                         mkl    defaults
blaze                     0.11.3           py35ha2c9137_0    defaults
bleach                    1.5.0                    py35_0    defaults
blosc                     1.14.3               hdbcaa40_0    defaults
bokeh                     0.12.13          py35h2f9c1c0_0    defaults
boost                     1.65.1                   py35_4    defaults
boto                      2.48.0           py35h2cfd601_1    defaults
bottleneck                1.2.1            py35he1b16f1_0    defaults
bzip2                     1.0.6                h9a117a8_4    defaults
ca-certificates           2019.1.23                     0    defaults
cairo                     1.14.12              h77bcde2_0    defaults
certifi                   2018.8.24                py35_1    defaults
cffi                      1.11.4           py35h9745a5d_0    defaults
chardet                   3.0.4            py35hb6e9ddf_1    defaults
click                     6.7              py35h353a69f_0    defaults
cloudpickle               0.5.2                    py35_1    defaults
clyent                    1.2.2            py35h491ffcb_1    defaults
cmake                     3.9.4                h142f0e9_0    defaults
colorama                  0.3.9            py35h81e2b6c_0    defaults
contextlib2               0.5.5            py35h6690dba_0    defaults
cryptography              2.1.4            py35hbeb2da1_0    defaults
cudatoolkit               8.0                           3    defaults
cudnn                     7.1.3                 cuda8.0_0    defaults
cupti                     8.0.61                        0    defaults
curl                      7.58.0               h84994c4_0    defaults
cycler                    0.10.0           py35hc4d5149_0    defaults
cython                    0.27.3           py35h6cdc64b_0    defaults
cytoolz                   0.9.0            py35h14c3975_0    defaults
dask                      0.16.1                   py35_0    defaults
dask-core                 0.16.1                   py35_0    defaults
datashape                 0.5.4            py35h5b068ae_0    defaults
dbus                      1.12.2               hc3f9b76_1    defaults
decorator                 4.2.1                    py35_0    defaults
distributed               1.20.2                   py35_0    defaults
dlib                      19.12.0                  pypi_0    pypi
docopt                    0.6.2                    pypi_0    pypi
docutils                  0.14             py35hd11081d_0    defaults
entrypoints               0.2.3            py35h48174a2_2    defaults
et_xmlfile                1.0.1            py35hde8d43e_0    defaults
expat                     2.2.5                he0dffb1_0    defaults
fastcache                 1.0.2            py35h14c3975_2    defaults
ffmpeg                    3.1.3                         0    menpo
filelock                  2.0.13           py35h4160c53_0    defaults
flask                     0.12.2           py35h679e90e_0    defaults
flask-cors                3.0.3            py35h9b2065d_0    defaults
fontconfig                2.12.6               h49f89f6_0    defaults
freetype                  2.8                  hab7d2ae_1    defaults
fribidi                   1.0.2                h14c3975_0    defaults
gast                      0.2.0                    py35_0    defaults
get_terminal_size         1.0.0                haa9412d_0    defaults
gevent                    1.2.2            py35he064abf_0    defaults
glib                      2.53.6               h5d9569c_2    defaults
glob2                     0.6              py35hff66265_0    defaults
gmp                       6.1.2                h6c8ec71_1    defaults
gmpy2                     2.0.8            py35hd0a1c9a_2    defaults
graphite2                 1.3.10               hf63cedd_1    defaults
greenlet                  0.4.12           py35h2547b41_0    defaults
grpcio                    1.12.1           py35hdbcaa40_0    defaults
gst-plugins-base          1.12.4               h33fb286_0    defaults
gstreamer                 1.12.4               hb53b477_0    defaults
h5py                      2.7.1            py35ha1a889d_0    defaults
harfbuzz                  1.7.6                hc5b324e_0    defaults
hdf5                      1.8.18               h6792536_1    defaults
heapdict                  1.0.0                    py35_2    defaults
html5lib                  0.9999999                py35_0    defaults
icu                       58.2                 h9c2bf20_1    defaults
idna                      2.6              py35h8605a33_1    defaults
imageio                   2.2.0            py35hd0a6de2_0    defaults
imagesize                 0.7.1            py35hf008fae_0    defaults
imutils                   0.3.6                    py35_0    mlgill
intel-openmp              2018.0.0             hc7b2577_8    defaults
ipykernel                 4.8.0                    py35_0    defaults
ipython                   6.2.1            py35hd850d2a_1    defaults
ipython_genutils          0.2.0            py35hc9e07d0_0    defaults
ipywidgets                7.1.1                    py35_0    defaults
isort                     4.2.15           py35h44a00d2_0    defaults
itsdangerous              0.24             py35h7c46880_1    defaults
jbig                      2.1                  hdba287a_0    defaults
jdcal                     1.3              py35hf1f7175_0    defaults
jedi                      0.11.1                   py35_0    defaults
jinja2                    2.10             py35h480ab6d_0    defaults
joblib                    0.13.2                   pypi_0    pypi
jpeg                      9b                   h024ee3a_2    defaults
jsonschema                2.6.0            py35h4395190_0    defaults
jupyter                   1.0.0                    py35_4    defaults
jupyter_client            5.2.2                    py35_0    defaults
jupyter_console           5.2.0            py35h4044a63_1    defaults
jupyter_core              4.4.0            py35ha89e94b_0    defaults
jupyterlab                0.31.5                   py35_0    defaults
jupyterlab_launcher       0.10.2                   py35_0    defaults
keras                     2.0.8                    py35_1    https://www.idiap.ch/software/bob/conda
kiwisolver                1.0.1            py35hcb1117a_0    defaults
lazy-object-proxy         1.3.1            py35h4c720c6_0    defaults
libboost                  1.65.1               habcd387_4    defaults
libcurl                   7.58.0               h1ad7b7a_0    defaults
libedit                   3.1                  heed3624_0    defaults
libffi                    3.2.1                hd88cf55_4    defaults
libgcc-ng                 7.2.0                h7cc24e2_2    defaults
libgfortran               3.0.0                         1    defaults
libgfortran-ng            7.2.0                h9f7466a_2    defaults
libpng                    1.6.34               hb9fc6fc_0    defaults
libprotobuf               3.5.2                h6f1eeef_0    defaults
libsodium                 1.0.15               hf101ebd_0    defaults
libssh2                   1.8.0                h9cfc8f7_4    defaults
libstdcxx-ng              7.2.0                h7a57d05_2    defaults
libtiff                   4.0.9                he85c1e1_1    defaults
libtool                   2.4.6                h544aabb_3    defaults
libuv                     1.14.0               h56b52c2_0    defaults
libxcb                    1.12                 hcd93eb1_4    defaults
libxml2                   2.9.7                h26e45fe_0    defaults
libxslt                   1.1.32               h1312cb7_0    defaults
llvmlite                  0.21.0           py35hcfd7307_0    defaults
locket                    0.2.0            py35h170bc82_1    defaults
lxml                      4.1.1            py35hf71bdeb_1    defaults
lzo                       2.10                 h49e0be7_2    defaults
markdown                  2.6.11                   py35_0    defaults
markupsafe                1.0              py35h4f4fcf6_1    defaults
matplotlib                2.2.2            py35h0e671d2_0    defaults
mccabe                    0.6.1            py35hd8c1f23_1    defaults
mistune                   0.8.3                    py35_0    defaults
mkl                       2018.0.2                      1    defaults
mkl-service               1.1.2                    py35_3    defaults
mkl_fft                   1.0.1            py35h3010b51_0    defaults
mkl_random                1.0.1            py35h629b387_0    defaults
moviepy                   0.2.3.4                  pypi_0    pypi
mpc                       1.0.3                hec55b23_5    defaults
mpfr                      3.1.5                h11a74b3_2    defaults
mpmath                    1.0.0            py35h7ce6e34_2    defaults
msgpack-python            0.5.1            py35h6bb024c_0    defaults
mtcnn                     0.0.7                    pypi_0    pypi
multipledispatch          0.4.9            py35h2ff591a_0    defaults
munkres                   1.0.12                   pypi_0    pypi
nbconvert                 5.3.1            py35hc5194e3_0    defaults
nbformat                  4.4.0            py35h12e6e07_0    defaults
ncurses                   6.0                  h9df7e31_2    defaults
networkx                  2.1                      py35_0    defaults
ninja                     1.8.2            py35h6bb024c_1    defaults
nltk                      3.2.5            py35h09ad193_0    defaults
nose                      1.3.7            py35hdc64897_2    defaults
notebook                  5.4.0                    py35_0    defaults
numba                     0.36.2          np113py35ha7cad2d_0    defaults
numexpr                   2.6.5            py35h7bf3b9c_0    defaults
numpy                     1.14.3           py35hcd700cb_1    defaults
numpy-base                1.14.3           py35h9be14a7_1    defaults
numpydoc                  0.7.0            py35he089a57_0    defaults
odo                       0.5.1            py35h102315f_0    defaults
olefile                   0.45.1                   py35_0    defaults
opencv                    3.4.1            py35h40b0b35_1    defaults
opencv-contrib-python     3.4.1.15                 pypi_0    pypi
openpyxl                  2.4.10                   py35_0    defaults
openssl                   1.0.2p               h14c3975_0    defaults
packaging                 16.8             py35h2260b46_1    defaults
pandas                    0.21.1           py35h20b78c2_0    defaults
pandoc                    1.19.2.1             hea2e7c5_1    defaults
pandocfilters             1.4.2            py35h1565a15_1    defaults
pango                     1.42.0               h377f3fa_0    defaults
parso                     0.1.1            py35h1b200a3_0    defaults
partd                     0.3.8            py35h68187f2_0    defaults
patchelf                  0.9                  hf79760b_2    defaults
path.py                   10.5             py35h39c98c1_0    defaults
pathlib2                  2.3.0            py35hd637de4_0    defaults
patsy                     0.5.0                    py35_0    defaults
pcre                      8.41                 hc27e229_1    defaults
pep8                      1.7.1                    py35_0    defaults
pexpect                   4.3.1                    py35_0    defaults
pickleshare               0.7.4            py35hd57304d_0    defaults
pillow                    5.1.0            py35h3deb7b8_0    defaults
pip                       19.0.3                   pypi_0    pypi
pixman                    0.34.0               hceecf20_3    defaults
pkginfo                   1.4.1            py35hb526c55_1    defaults
pluggy                    0.6.0            py35hb428803_0    defaults
ply                       3.10             py35h459f1a0_0    defaults
prompt_toolkit            1.0.15           py35hc09de7a_0    defaults
protobuf                  3.5.2            py35hf484d3e_0    defaults
psutil                    5.4.3            py35h14c3975_0    defaults
ptyprocess                0.5.2            py35h38ce0a3_0    defaults
py                        1.5.2            py35hfe8c5ce_0    defaults
py-boost                  1.65.1           py35hf484d3e_4    defaults
pyannote-algorithms       0.8                      pypi_0    pypi
pyannote-core             1.3.1                    pypi_0    pypi
pyannote-parser           0.7.1                    pypi_0    pypi
pyannote-video            1.6                      pypi_0    pypi
pycodestyle               2.3.1            py35h8d2105a_0    defaults
pycosat                   0.6.3            py35h6b6bb97_0    defaults
pycparser                 2.18             py35h61b3040_1    defaults
pycrypto                  2.6.1            py35h14c3975_7    defaults
pycurl                    7.43.0.1         py35hb7f436b_0    defaults
pyflakes                  1.6.0            py35hd1da72e_0    defaults
pygments                  2.2.0            py35h0f41973_0    defaults
pylint                    1.8.2                    py35_0    defaults
pyodbc                    4.0.22           py35hf484d3e_0    defaults
pyopenssl                 17.5.0           py35h4f8b8c8_0    defaults
pyparsing                 2.2.0            py35h041ed72_1    defaults
pyqt                      5.6.0            py35h0e41ada_5    defaults
pysocks                   1.6.7            py35h6aefbb0_1    defaults
pytables                  3.4.3            py35h68a8fdc_2    defaults
pytest                    3.3.2                    py35_0    defaults
python                    3.5.5                hc3d631a_1    defaults
python-dateutil           2.6.1            py35h90d5b31_1    defaults
pytorch-cpu               0.4.0                py35_cpu_1    pytorch
pytz                      2017.3           py35hb13c558_0    defaults
pywavelets                0.5.2            py35h53ec731_0    defaults
pyyaml                    3.12             py35h46ef4ae_1    defaults
pyzmq                     16.0.3           py35ha889422_0    defaults
qt                        5.6.2               hd25b39d_14    defaults
qtawesome                 0.4.4            py35hd68ff89_0    defaults
qtconsole                 4.3.1            py35h4626a06_0    defaults
qtpy                      1.3.1            py35h881af26_0    defaults
readline                  7.0                  ha6073c6_4    defaults
requests                  2.18.4           py35hb9e6ad1_1    defaults
rhash                     1.3.5                hbf7ad62_1    defaults
rope                      0.10.7           py35h986144c_0    defaults
ruamel_yaml               0.15.35          py35h14c3975_1    defaults
scikit-image              0.13.0           py35h3573165_1    defaults
scikit-learn              0.19.1           py35hbf1f462_0    defaults
scipy                     1.1.0            py35hfc37229_0    defaults
seaborn                   0.8.1            py35h04cba02_0    defaults
send2trash                1.4.2                    py35_0    defaults
setuptools                40.2.0                   py35_0    defaults
simplegeneric             0.8.1                    py35_2    defaults
simplejson                3.13.2                   pypi_0    pypi
singledispatch            3.4.0.3          py35h0cd4ec3_0    defaults
sip                       4.18                     py35_0    defaults
six                       1.11.0           py35h423b573_1    defaults
snappy                    1.1.7                hbae5bb6_3    defaults
snowballstemmer           1.2.1            py35h5435977_0    defaults
sortedcollections         1.1.2                    pypi_0    pypi
sortedcontainers          2.0.4                    pypi_0    pypi
sphinx                    1.6.6                    py35_0    defaults
sphinxcontrib             1.0              py35hcb7efd2_1    defaults
sphinxcontrib-websupport  1.0.1            py35h3537809_1    defaults
spyder                    3.2.6                    py35_0    defaults
sqlalchemy                1.2.1            py35h14c3975_0    defaults
sqlite                    3.22.0               h1bed415_0    defaults
statsmodels               0.8.0            py35haa9d50b_0    defaults
sympy                     1.1.1            py35h919b29a_0    defaults
tblib                     1.3.2            py35hf1eb0b4_0    defaults
tensorboard               1.7.0            py35hf484d3e_1    defaults
tensorflow                1.4.1                         0    defaults
tensorflow-base           1.4.1            py35hd00c003_2    defaults
tensorflow-gpu            1.7.0                         0    defaults
tensorflow-gpu-base       1.7.0            py35h5b7bae4_1    defaults
tensorflow-tensorboard    1.5.1            py35hf484d3e_1    defaults
termcolor                 1.1.0                    py35_1    defaults
terminado                 0.8.1                    py35_1    defaults
testpath                  0.3.1            py35had42eaf_0    defaults
tk                        8.6.7                hc745277_3    defaults
toolz                     0.9.0                    py35_0    defaults
torchvision-cpu           0.2.1                    py35_1    pytorch
tornado                   4.5.3                    py35_0    defaults
tqdm                      4.23.4                   pypi_0    pypi
traitlets                 4.3.2            py35ha522a97_0    defaults
typing                    3.6.2            py35hcadae7e_0    defaults
unicodecsv                0.14.1           py35h4805c0c_0    defaults
unixodbc                  2.3.4                hc36303a_1    defaults
urllib3                   1.22             py35h2ab6e29_0    defaults
wcwidth                   0.1.7            py35hcd08066_0    defaults
webencodings              0.5.1            py35hb6cf162_1    defaults
werkzeug                  0.14.1                   py35_0    defaults
wheel                     0.31.1                   py35_0    defaults
widgetsnbextension        3.1.0                    py35_0    defaults
wrapt                     1.10.11          py35hfdafd39_0    defaults
xarray                    0.10.0                   pypi_0    pypi
xlrd                      1.1.0            py35h45a0a2a_1    defaults
xlsxwriter                1.0.2            py35h8ee9321_0    defaults
xlwt                      1.3.0            py35h884a0cf_0    defaults
xz                        5.2.3                h55aa19d_2    defaults
yaml                      0.1.7                had09818_2    defaults
zeromq                    4.2.2                hbedb6e5_2    defaults
zict                      0.1.3            py35h29275ca_0    defaults
zlib                      1.2.11               ha838bed_2    defaults

</details>
